mod.go

```diff
@@ -0,0 +1,669 @@
+// uint256: Fixed size 256-bit math library
+// Copyright 2021 uint256 Authors
+// SPDX-License-Identifier: BSD-3-Clause
+
+package uint256
+
+import (
+	"math/bits"
+	"sync"
+)
+
+// Cache for reciprocal()
+//
+// Each cache set contains its own mutex to reduce lock contention in highly
+// multithreaded settings.
+//
+// Adjust cacheIndexBits and cacheWays to scale the size of the cache.
+// Reasonable values are quite small, e.g. cacheIndexBits from 2 to 10, and
+// cacheWays from 3 to 11 (3+8n makes set size an integer number of cachelines)
+
+const (
+	cacheIndexBits = 9
+	cacheWays = 3
+
+	cacheSets = 1 << cacheIndexBits
+	cacheMask = cacheSets - 1
+)
+
+type cacheSet struct {
+	hit	uint64
+	miss	uint64
+
+	rw	sync.RWMutex
+	mod	[cacheWays]Int
+	inv	[cacheWays][5]uint64
+}
+
+var cache [cacheSets]cacheSet
```

Comment:
I was wondering why the cache does not show up as allocs on the benchmark stats, but now I see: they're statically allocated. So the whole cache is `512` elements of `256` bytes each, and takes up a fixed chunk of 128K memory.

Reply:
Yes, that is correct.